src/com/lftechnology/java/training/krishna/employeemanagement/dao/employee/impl/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,291 @@
+			preparedStatement.setString(3, entity.getFullname());
+			preparedStatement.setString(4, entity.getDepartment());
+			preparedStatement.setString(5, entity.getAddress());
+			preparedStatement.setString(6, entity.getRole());
+			preparedStatement.setInt(7, entity.getIsTerminated());
+			preparedStatement.setDate(8, entity.getCreatedAt());
+			int affectedRows = preparedStatement.executeUpdate();
+			if (affectedRows == 0) {
+				throw new SQLException(
+					"Creating user failed, no rows affected.");
+			}
+			ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+			if (generatedKeys.next()) {
+				entity.setId(generatedKeys.getInt(1));
+			}
+		}
+		catch (SQLException e) {
+			LOGGER.log(
+				Level.SEVERE, ConstantUtils.EXCEPTION_STATEMENT, e);
+		}
+		finally {
+			try {
+				preparedStatement.close();
+			}
+			catch (SQLException e) {
+				LOGGER.log(Level.SEVERE, ConstantUtils.ERROR_STATEMENT, e);
+			}
+		}
+		return entity;
+	}
+
+	@Override
+	public List<Employee> findAll(String key) {
+
+		List<Employee> empList = new ArrayList<Employee>();
+		try {
+			preparedStatement = connection.prepareStatement(SEARCH_USER_QUERY);
+			preparedStatement.setString(1, "%" + key + "%");
+			preparedStatement.setString(2, "%" + key + "%");
+			preparedStatement.setString(3, "%" + key + "%");
+			ResultSet resultSet = preparedStatement.executeQuery();
+			while (resultSet.next()) {
+				Employee emp = new Employee();
+				emp.setUsername(resultSet.getString("username"));
+				emp.setFullname(resultSet.getString("fullname"));
+				emp.setDepartment(resultSet.getString("department"));
+				emp.setAddress(resultSet.getString("address"));
+				emp.setRole(resultSet.getString("role"));
+				empList.add(emp);
+			}
+		}
+		catch (SQLException e) {
+			LOGGER.log(
+				Level.SEVERE,  ConstantUtils.EXCEPTION_STATEMENT, e);
+		}
+		finally {
+			try {
+				preparedStatement.close();
+			}
+			catch (SQLException e) {
+				LOGGER.log(Level.SEVERE, ConstantUtils.ERROR_STATEMENT, e);
+			}
+		}
+		return empList;
+	}
+
+	@Override
+	public Employee findById(Integer id) {
+
+		Employee emp = new Employee();
+		try {
+			preparedStatement = connection.prepareStatement(SEARCH_BY_ID_QUERY);
+			preparedStatement.setInt(1, id);
+			ResultSet resultSet = preparedStatement.executeQuery();
+			while (resultSet.next()) {
+
+				emp.setUsername(resultSet.getString("username"));
+				emp.setFullname(resultSet.getString("fullname"));
+				emp.setDepartment(resultSet.getString("department"));
+				emp.setAddress(resultSet.getString("address"));
+				emp.setRole(resultSet.getString("role"));
+			}
+		}
+		catch (SQLException e) {
+			LOGGER.log(
+				Level.SEVERE,  ConstantUtils.EXCEPTION_STATEMENT, e);
+		}
+		finally {
+			try {
+				preparedStatement.close();
+			}
+			catch (SQLException e) {
+				LOGGER.log(Level.SEVERE, ConstantUtils.ERROR_STATEMENT, e);
+			}
+		}
+		return emp;
+	}
+
+	@Override
+	public Employee update(Employee entity) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "update" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)